### PR TITLE
refactor: move BDB (bitdb / db.h) interaction from init.cpp to wallet.cpp

### PR DESCRIPTION
**Pull request by jonasschnelli**
this will remove db.h from init.cpp

### CODE REVIEW

**Comment by jtimon** [comment]
ut ACK

**Comment by jgarzik** [comment]
ut ACK

**Comment by laanwj** [comment]
Nice! Going to test.

File: src/init.cpp [hunk 1]
```diff
@@ -835,47 +834,17 @@ bool AppInit2(boost::thread_group& threadGroup)
...
-                string msg = strprintf(_("Warning: wallet.dat corrupt, data salvaged!"
-                                         " Original wallet.dat saved as wallet.{timestamp}.bak in %s; if"
-                                         " your balance or transactions are incorrect you should"
-                                         " restore from a backup."), strDataDir);
-                InitWarning(msg);
-            }
-            if (r == CDBEnv::RECOVER_FAIL)
-                return InitError(_("wallet.dat corrupt, salvage failed"));
-        }
+        string warningString;
```
Comment (laanwj) [inline src/init.cpp]: Nit: please use `std::string`. We'd like to get rid of the global `using namespace`s at some point.

Comment (jonasschnelli) [inline src/init.cpp]: Okay. Updated. There are multiple string instantiations without the `std::` namespace within init.cpp. I don't touch these (as well as the one i code-moved) for now.

**Comment by laanwj** [comment]
Merged via 2bb1c87700a80f0b20b4b2cbd68a0d82a18dbcc0 (trivial header conflict fixed)

**Pull request closed without merging**